internal/provider/models/deployment.go

```diff
@@ -11,7 +11,66 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-type Deployment struct {
+type DeploymentResource struct {
+	// Common fields
+	Id                          types.String `tfsdk:"id"`
+	Name                        types.String `tfsdk:"name"`
+	Description                 types.String `tfsdk:"description"`
+	CreatedAt                   types.String `tfsdk:"created_at"`
+	UpdatedAt                   types.String `tfsdk:"updated_at"`
+	CreatedBy                   types.Object `tfsdk:"created_by"`
+	UpdatedBy                   types.Object `tfsdk:"updated_by"`
+	WorkspaceId                 types.String `tfsdk:"workspace_id"`
+	Type                        types.String `tfsdk:"type"`
+	Region                      types.String `tfsdk:"region"`
+	CloudProvider               types.String `tfsdk:"cloud_provider"`
+	OriginalAstroRuntimeVersion types.String `tfsdk:"original_astro_runtime_version"`
+	AstroRuntimeVersion         types.String `tfsdk:"astro_runtime_version"`
+	AirflowVersion              types.String `tfsdk:"airflow_version"`
+	Namespace                   types.String `tfsdk:"namespace"`
+	ContactEmails               types.Set    `tfsdk:"contact_emails"`
+	Executor                    types.String `tfsdk:"executor"`
+	SchedulerCpu                types.String `tfsdk:"scheduler_cpu"`
+	SchedulerMemory             types.String `tfsdk:"scheduler_memory"`
+	SchedulerAu                 types.Int64  `tfsdk:"scheduler_au"`
+	SchedulerReplicas           types.Int64  `tfsdk:"scheduler_replicas"`
+	ImageTag                    types.String `tfsdk:"image_tag"`
+	ImageRepository             types.String `tfsdk:"image_repository"`
+	ImageVersion                types.String `tfsdk:"image_version"`
+	EnvironmentVariables        types.Set    `tfsdk:"environment_variables"`
+	WebserverIngressHostname    types.String `tfsdk:"webserver_ingress_hostname"`
+	WebserverUrl                types.String `tfsdk:"webserver_url"`
+	WebserverAirflowApiUrl      types.String `tfsdk:"webserver_airflow_api_url"`
+	Status                      types.String `tfsdk:"status"`
+	StatusReason                types.String `tfsdk:"status_reason"`
+	DagTarballVersion           types.String `tfsdk:"dag_tarball_version"`
+	DesiredDagTarballVersion    types.String `tfsdk:"desired_dag_tarball_version"`
+	IsCicdEnforced              types.Bool   `tfsdk:"is_cicd_enforced"`
+	IsDagDeployEnabled          types.Bool   `tfsdk:"is_dag_deploy_enabled"`
+	WorkloadIdentity            types.String `tfsdk:"workload_identity"`
+	ExternalIps                 types.Set    `tfsdk:"external_ips"`
+	OidcIssuerUrl               types.String `tfsdk:"oidc_issuer_url"`
+	WorkerQueues                types.Set    `tfsdk:"worker_queues"`
+
+	// Hybrid and dedicated specific fields
+	ClusterId types.String `tfsdk:"cluster_id"`
+
+	// Hybrid deployment specific fields
+	TaskPodNodePoolId types.String `tfsdk:"task_pod_node_pool_id"`
+
+	// Hosted deployment specific fields
```

Comment:
nit: since you mentioned dedicated above, so that we don't have to infer what you mean by hosted below
```suggestion
	// Hosted (shared and dedicated) deployment specific fields
```

Reply:
Non-blocking

Reply:
will add